DNN Platform/Library/Services/ClientDependency/DependencyController.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Services.ClientDependency
+{
+    using System;
+    using System.Collections;
+    using System.Web;
+    using System.Web.UI;
+
+    using DotNetNuke.Abstractions.ClientResources;
+    using Microsoft.Extensions.DependencyInjection;
+
+    internal class DependencyController
+    {
+        internal static IClientResourceController GetClientResourcesController(Page page)
+        {
+            var serviceProvider = GetCurrentServiceProvider(page.Request.RequestContext.HttpContext);
```

Comment:
Maybe Globals.GetCurrentServiceProvider() can do the job in a more uniform way inside dnn

Reply:
This is a pattern we're actively trying to discourage, it should only be used for backwards compatibility reasons. That's why [`Globals.GetCurrentServiceProvider` is `internal`](https://github.com/dnnsoftware/Dnn.Platform/blob/6267a7ba78028ccc8645f5b94a8bc165ace10b30/DNN%20Platform/Library/Common/Globals.cs#L3541-L3546)